llvm/lib/Target/X86/X86InstrAVX10.td

```diff
@@ -417,3 +417,173 @@ defm VMINMAXSH : avx10_minmax_scalar<"vminmaxsh", v8f16x_info, X86vminmaxs, X86v
+    defm Z256rrb : AVX512_maskable<Opc, MRMSrcReg, DestInfo.info256,
+                              (outs DestInfo.info256.RC:$dst),
+                              (ins SrcInfo.info256.RC:$src, AVX512RC:$rc),
+                              OpStr, "$rc, $src", "$src, $rc",
+                              (DestInfo.info256.VT (MaskNode
+                               SrcInfo.info256.RC:$src, (i32 timm:$rc)))>,
+                              Sched<[sched.YMM]>, EVEX, EVEX_RC, EVEX_B;
+    }
+}
+
+// Conversion with SAE
+multiclass
+    avx10_sat_cvt_sae<bits<8> Opc, string OpStr, X86SchedWriteWidths sched,
+                      AVX512VLVectorVTInfo DestInfo, AVX512VLVectorVTInfo SrcInfo,
+                      SDNode Node> {
+    let Uses = [MXCSR] in
+    defm Zrrb : AVX512_maskable<Opc, MRMSrcReg, DestInfo.info512,
+                              (outs DestInfo.info512.RC:$dst),
+                              (ins SrcInfo.info512.RC:$src),
+                              OpStr, "{sae}, $src", "$src, {sae}",
+                              (DestInfo.info512.VT (Node SrcInfo.info512.RC:$src))>,
+                              Sched<[sched.ZMM]>, EVEX, EVEX_B;
+    let Predicates = [HasAVX10_2], hasEVEX_U = 1 in {
+    defm Z256rrb : AVX512_maskable<Opc, MRMSrcReg, DestInfo.info256,
+                              (outs DestInfo.info256.RC:$dst),
+                              (ins SrcInfo.info256.RC:$src),
+                              OpStr, "{sae}, $src", "$src, {sae}",
+                              (DestInfo.info256.VT (Node SrcInfo.info256.RC:$src))>,
+                              Sched<[sched.YMM]>, EVEX, EVEX_B;
+    }
+}
+
+multiclass avx10_sat_cvt_base<bits<8> Opc, string OpStr, X86SchedWriteWidths sched,
+                               SDNode MaskNode, AVX512VLVectorVTInfo DestInfo,
+                               AVX512VLVectorVTInfo SrcInfo> {
+  let Predicates = [HasAVX10_2_512] in
+  defm Z : avx10_sat_cvt_rmb<Opc, OpStr, sched.ZMM,
+           DestInfo.info512, SrcInfo.info512,
+           MaskNode>,
+      EVEX, EVEX_V512;
+  let Predicates = [HasAVX10_2] in {
+    defm Z256
+        : avx10_sat_cvt_rmb<Opc, OpStr, sched.YMM,
+           DestInfo.info256, SrcInfo.info256,
+          MaskNode>,
+          EVEX, EVEX_V256;
+    defm Z128
+        : avx10_sat_cvt_rmb<Opc, OpStr, sched.XMM,
+          DestInfo.info128, SrcInfo.info128,
+          MaskNode>,
+          EVEX, EVEX_V128;
+  }
+}
+
+defm VCVTNEBF162IBS : avx10_sat_cvt_base<0x69, "vcvtnebf162ibs",
+                      SchedWriteVecIMul, X86vcvtnebf162ibs,
+                      avx512vl_i16_info, avx512vl_bf16_info>,
+                      AVX512XDIi8Base, T_MAP5, EVEX_CD8<16, CD8VF>;
+defm VCVTNEBF162IUBS : avx10_sat_cvt_base<0x6b, "vcvtnebf162iubs",
+                      SchedWriteVecIMul, X86vcvtnebf162iubs,
+                      avx512vl_i16_info, avx512vl_bf16_info>,
+                      AVX512XDIi8Base, T_MAP5, EVEX_CD8<16, CD8VF>;
```

Comment:
This can be
```
defm VCVTNEBF162IBS : avx10_sat_cvt_base<0x69, "vcvtnebf162ibs",
                      SchedWriteVecIMul, X86vcvtnebf162ibs,
                      avx512vl_i16_info, avx512vl_bf16_info>,
                      avx10_sat_cvt_base<0x6b, "vcvtnebf162iubs",
                      SchedWriteVecIMul, X86vcvtnebf162iubs,
                      avx512vl_i16_info, avx512vl_bf16_info>,
                      AVX512XDIi8Base, T_MAP5, EVEX_CD8<16, CD8VF>;
```

The same below.

Reply:
how to extra `U` to `defm VCVTNEBF162IBS`?

Reply:
Oh, I commented on wrong instruction. I meant
```
defm VCVTPH2IBS : avx10_sat_cvt_base<0x69, "vcvtph2ibs", SchedWriteVecIMul,
                                     X86vcvtph2ibs, avx512vl_i16_info, avx512vl_f16_info>,
                  avx10_sat_cvt_rc<0x69, "vcvtph2ibs", SchedWriteVecIMul,
                                   avx512vl_i16_info, avx512vl_f16_info,
                                   X86vcvtph2ibsRnd>,
                  AVX512PSIi8Base, T_MAP5, EVEX_CD8<16, CD8VF>;
```

Reply:
aa5cc99